tests/page/page-screenshot.spec.ts

```diff
@@ -339,3 +339,247 @@ it.describe('page screenshot', () => {
+      expect(screenshot1.equals(screenshot2)).toBe(false);
+    });
+
+    // Make a screenshot that finishes all finite animations.
+    const screenshot1 = await div.screenshot({
+      disableAnimations: true,
+    });
+    await raf(page);
+    // Make sure finite transition is not restarted.
+    const screenshot2 = await div.screenshot();
+    expect(screenshot1.equals(screenshot2)).toBe(true);
+
+    expect(await page.evaluate(() => window['__TRANSITION_END'])).toBe(true);
+  });
+
+  it('should capture screenshots after layoutchanges in transitionend event', async ({ page, server }) => {
+    await page.goto(server.PREFIX + '/css-transition.html');
+    const div = page.locator('div');
+    await div.evaluate(el => {
+      el.addEventListener('transitionend', () => {
+        const time = Date.now();
+        // Block main thread for 200ms, emulating heavy layout.
+        while (Date.now() - time < 200) ;
+        const h1 = document.createElement('h1');
+        h1.textContent = 'woof-woof';
+        document.body.append(h1);
+      }, false);
+    });
+
+    await it.step('make sure transition is actually running', async () => {
+      const screenshot1 = await page.screenshot();
+      await raf(page);
+      const screenshot2 = await page.screenshot();
+      expect(screenshot1.equals(screenshot2)).toBe(false);
+    });
+
+    // 1. Make a screenshot that finishes all finite animations
+    //    and triggers layout.
+    const screenshot1 = await page.screenshot({
+      disableAnimations: true,
+    });
+
+    // 2. Make a second screenshot after h1 is on screen.
+    await expect(page.locator('h1')).toBeVisible();
+    await expect(page.locator('h1')).toHaveText('woof-woof');
+    const screenshot2 = await page.screenshot();
+
+    // 3. Make sure both screenshots are equal, meaning that
+    //    first screenshot actually was taken after transitionend
+    //    changed layout.
+    expect(screenshot1.equals(screenshot2)).toBe(true);
+  });
+
+  it('should not change animation with playbackRate equal to 0', async ({ page, server }) => {
+    await page.goto(server.PREFIX + '/rotate-z.html');
+    await page.evaluate(async () => {
+      window.animation = document.getAnimations()[0];
+      window.animation.updatePlaybackRate(0);
+      await window.animation.ready;
+      window.animation.currentTime = 500;
+    });
+    const screenshot1 = await page.screenshot({
+      disableAnimations: true,
+    });
+    await raf(page);
+    const screenshot2 = await page.screenshot({
+      disableAnimations: true,
+    });
+    expect(screenshot1.equals(screenshot2)).toBe(true);
+    expect(await page.evaluate(() => ({
+      playbackRate: window.animation.playbackRate,
+      currentTime: window.animation.currentTime,
+    }))).toEqual({
+      playbackRate: 0,
+      currentTime: 500,
+    });
+  });
+
+  it('should trigger particular events for finite animation', async ({ page, server }) => {
```

Comment:
Let's copy this test with a finite css animation instead.

Reply:
done